src/contrasts.jl

```diff
@@ -520,10 +558,29 @@ check_contrasts_size(mat::Matrix, n_lev::Int) =
                         "Expected $((n_lev, n_lev-1)), got $(size(mat))"))
 
 ## constructor with optional keyword arguments, defaulting to nothing
-ContrastsCoding(mat::Matrix; base=nothing, levels=nothing) =
-    ContrastsCoding(mat, base, levels)
+ContrastsCoding(mat::AbstractMatrix; levels=nothing) =
+    ContrastsCoding(mat, levels)
 
 function contrasts_matrix(C::ContrastsCoding, baseind, n)
     check_contrasts_size(C.mat, n)
     C.mat
 end
+
+## hypothesis matrix
+needs_intercept(mat) =
+    (rank(mat) < size(mat, 1)) &&
+    !all(mapslices(sum, mat, dims=1) .< 10eps(eltype(mat)))
```

Comment:
IIRC `mapslices` makes a copy of each slice, so better use `eachslice`.
```suggestion
    !all(sum(row) < 10eps(eltype(mat)) for row in eachrow(x))
```